src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -279,12 +282,14 @@ private static void checkProperThreadForStateMutation()
     {
         public void run()
         {
+            boolean locked = false;
             try
             {
                 //wait on messaging service to start listening
                 MessagingService.instance().waitUntilListening();
 
-                taskLock.lock();
```

Comment:
here the lock is called in a try block, after some method which may throw - in such case we could attempt to unlock a lock which was not acquired (or was acquired somewhere else in the meantime